libusb/os/windows_winusb.c

```diff
@@ -2017,6 +2017,188 @@ static int winusb_get_device_list(struct libusb_context *ctx, struct discovered_
 	return r;
 }
 
+#ifdef _MSC_VER
+#pragma pack(push, 1)
+#endif
+
+struct string_descriptor_s {
+	UCHAR bLength;
+	UCHAR bDescriptorType;
+	uint8_t bString[252];  // UTF-16LE, must be even length
+} LIBUSB_PACKED;
+
+#ifdef _MSC_VER
+#pragma pack(pop)
+#endif
+
+struct string_descriptor_req_s {
+	USB_DESCRIPTOR_REQUEST req;
+	struct string_descriptor_s desc;
+};
+
+/*
+ * \brief Convert a UTF-16 little endian string to a UTF-8 string.
+ *
+ * \param src The input UTF-16 little endian string.
+ * \param src_length The length of the src buffer in bytes which
+ *		may or may not include the null terminator.
+ * \param dst The output UTF-8 string.
+ * \param dst_length The length of dst buffer in bytes, which should
+ *		be large enough to fit the string including null terminator.
+ * \return The total number of bytes in the UTF-8 string, including
+ *      the null terminator.
+ *      If this is greater than dst_length, then dst is truncated.
+ * \note From MiniBitty RTOS & framework, currently closed source.
+ */
+static int usbi_utf16le_to_utf8(uint8_t const* src, int src_length, char* dst, int dst_length) {
+	size_t count = 0;
+	uint16_t codepoint;
+	bool overflow = false;
+
+	if (NULL == dst) {
+		dst_length = 0;
+	}
+	if (dst_length > 0) {
+		dst[0] = 0;  // empty string by default
+	}
+	if ((src == NULL) || (src_length <= 0)) {
+		return 0;
+	}
+
+	for (size_t k = 0; k < src_length; k += 2) {
+		codepoint = *src++;
+		codepoint |= ((uint16_t)*src++) << 8;
+		if (codepoint <= 0x7f) {
+			if (count < dst_length) {
+				dst[count] = (char)codepoint;
+			} else {
+				overflow = true;
+			}
+			count += 1;
+			if (0 == codepoint) {
+				--count;  // add null-terminator below
+				break;
+			}
+		}
+		else if (codepoint < 0x7ff) {
```

Comment:
Please check the MSVC build (e.g. [here](https://ci.appveyor.com/project/LudovicRousseau/libusb/builds/50314795/job/aqfsst24q0bclraj)). I believe with `/W4 /WX` this generates compilation error based on:
```
C4018: '<' : signed/unsigned mismatch 
```

Reply:
I have been building using the Visual Studio solution included with libusb, but I don't see these warnings.  Does the appveyor build use a different project?  Regardless, I have (hopefully) fixed this.